[feature] support multiple ovf_network mapping, enhance/fix upstream issue #103

lamtt77 opened the pull request:
This PR created from my actual use-cases involved multiple ovf network interfaces already defined in a signed ova file (by other).

Feature:
- Supports both single --network and/or multiple --net for network_interfaces creation with ova/ovf sources.  So multiple network interfaces already defined in ova/ovf source file will not raise error anymore.
- Before the changes, you have to extract ovf file from ova source and manually change to only single network interface and then re-package that ova file - which will not work if you have to deal with signed ova file.
- Please see original issue: https://github.com/josenk/terraform-provider-esxi/issues/103

The changes should work with any variations or orders of the following declaration:
```conf
  network_interfaces {
    virtual_network = "VM Network"
    nic_type        = "vmxnet3"
    ovf_network     = "WAN"
  }

  network_interfaces {
    virtual_network = "VyOS VLAN 10"
    nic_type        = "vmxnet3"
    ovf_network     = "LAN"
  }

  network_interfaces {
    virtual_network = "VyOS VLAN 20"
    nic_type        = "vmxnet3"
  }
```

In the above example, the ova source file defined 2 default ovf network interfaces (WAN and LAN), therefore ```--net:'WAN=VM Network' --net:'LAN=VM Network'``` will be generated as new 'net_param'.  Without these changes, you will see the error message "OVFtool error: No network mapping specified".

Support 'net_param' style:
- ovf network mapping: --net:'WAN=VM Network' --net:'LAN=VM Network'
- default style: --network='VM Network'
- mixed: --network='VyOS VLAN 20' --net:'WAN=VM Network' --net:'LAN=VyOS VLAN 10'

Test case 1:
- Create with the above example
- Add/update
``` conf
  network_interfaces {
    virtual_network = "VyOS VLAN 30"
    nic_type        = "vmxnet3"
  }

  network_interfaces {
    virtual_network = "VyOS VLAN 40"
    nic_type        = "vmxnet3"
  }
```
- Run terraform apply

Test case 2:
- Create with 1 single network
- Update with 2,3 more network interfces 
- Run terraform apply

Test case 3:
- Create with 4 mixed network interfces
- Delete 2 network interfaces
- Run terraform apply

Test case 4:
- Create with default (old) style single network
- Run terraform apply

Test case 5:
- Define multiple 'esxi_guest' with ovf_network and without ovf_network
- Run terraform apply

Please feel free to let me know should you need any changes/correction from this, I would be happy to revise accordingly.

josenk commented:
Thanks for the contribution.  I need to do some regression testing to make sure all is OK.    I'll update this pr if there's any issues.

lamtt77 commented:
I have fixed an issue when ovf_network's value had not been persisted correctly in terraform.tfstate and updated more test case.  Everything looks good now from my end.  Thanks!
